src/routes/teams/index.ts

```diff
@@ -0,0 +1,122 @@
+import { OpenAPIHono } from "@hono/zod-openapi";
+
+import prisma from "../../lib/prisma-client.js";
+import { getTeam, getTeamById, createTeam, deleteTeam } from "./routes.js";
+
+import { Role } from "@prisma/client";
+import { checkRole, getCurrentUser } from "../../lib/auth-provider.js";
+
+const teamRouter = new OpenAPIHono();
+
+teamRouter.openapi(createTeam, async (ctx) => {
+  try {
+    const user = await prisma.user.findUnique({
+      where: { id: getCurrentUser(ctx).id },
+    });
+    const allowedRoles: Role[] = ["SUPER_ADMIN", "ADMIN"];
+    const { name, imageId } = ctx.req.valid("json");
+
+    if (!checkRole(allowedRoles, ctx) && !user?.isLeader) {
+      return ctx.text("Unauthorized", 403);
+    }
+
+    await prisma.team.create({
+      data: {
+        name: name,
+        imageId: imageId,
+      },
+    });
+
+    return ctx.text("Team created successfully.", 200);
+  } catch (error) {
+    console.error("Error occurred", error);
+    return ctx.text("An unexpected error occurred", 500);
+  }
+});
+
+teamRouter.openapi(getTeam, async (ctx) => {
+  try {
+    if (checkRole([Role.USER], ctx)) {
+      return ctx.text("Unauthorized", 403);
+    }
+
+    const teams = await prisma.team.findMany();
+
+    if (!teams) {
```

Comment:
teams.length === 0 use this as findMany will give a empty array if no teams registered